dask/array/core.py

```diff
@@ -1573,9 +1573,479 @@ def __float__(self):
     def __complex__(self):
         return self._scalarfunc(complex)
 
+    def __index__(self):
+        return self._scalarfunc(int)
+
     def __setitem__(self, key, value):
         from .routines import where
 
+        def parse_indices(shape, indices):
```

Comment:
`dask.array.core` is already a really big module. If possible, it would be best to keep most of this logic/helper functions in a separate module -- maybe `slicing.py` or something new?

Reply:
That sounds like a good idea. Not being familiar with it, I was hoping that the library style would get imposed.

Reply:
These nested functions have been moved to `slicing.py`, and simplified a bit by reusing other functions in that module (such as `normailize_index`). See https://github.com/dask/dask/pull/7033/commits/c46ccb2e39bc520f0c8012b823307e823c0631f2